### PR TITLE
refactor AA page names and add tests

### PR DESCRIPTION
**Pull request by avertrees**
## Ticket:

- JIRA ticket [DR-3257](https://newyorkpubliclibrary.atlassian.net/browse/DR-3257)

## This PR does the following:

- refactors AA page name generation and adds tests

## Open Questions

- might be overkill but whatever

## How has this been tested? How should a reviewer test this?

- locally in the Adobe Analytics debugger chrome extension and with unit tests

## Accessibility concerns or updates

- N/A

### Checklist:

- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.
- [x] I have updated the CHANGELOG.md.

[DR-3257]: https://newyorkpubliclibrary.atlassian.net/browse/DR-3257?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/error.tsx [hunk 1]
```diff
@@ -19,7 +20,12 @@ export default function Error({
   }, [error]);
 
   return (
-    <PageLayout activePage="serverError">
+    <PageLayout
+      activePage="serverError"
+      adobeAnalyticsPageName={createAdobeAnalyticsPageName(
```
[the remark this answers is not in the document]
Comment (avertrees) [reply app/error.tsx:25]: no. while the error pages were falling under "undefined," I think something else was happening. After looking at the AA dashboard for the last few weeks, it looks like the /divisions/:slug pages send the correct page name to AA 99% of the time (not exact math but it's more than most). So, perhaps something else was happening? I checked the page name in the AA debugger and they were correct for all of the pages.

[the remark this answers is not in the document]
Comment (avertrees) [reply app/error.tsx:25]: also realizing now that I didn't check the page names with pagination... I'll do that

File: app/items/[uuid]/page.tsx [hunk 2]
```diff
@@ -43,7 +42,7 @@ export default async function ItemPage({ params }: ItemProps) {
           url: `/items/${params.uuid}`,
         },
       ]}
-      adobeAnalyticsPageName={pageName}
+      adobeAnalyticsPageName={createAdobeAnalyticsPageName("items", item.title)}
```
[the remark this answers is not in the document]
Comment (7emansell) [reply app/items/[uuid]/page.tsx:45]: As long as AA distinguishes between qa and prod views this seems fine to me

File: app/error.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (avertrees) [reply app/error.tsx:25]: pagenames work as expected with pagination in the debugger extension

**Review by EdwinGuzman — APPROVED** [review]

File: app/items/[uuid]/page.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (avertrees) [reply app/items/[uuid]/page.tsx:45]: QA DCF sends AA data to a different dashboard than prod, so I think this is fine

**Pull request merged**